harvester/harvester-installer
# Add monitoring integation

https://github.com/harvester/harvester/issues/576
Add built-in monitoring integration 

**What does this PR do:**
- Add built-in monitoring integration by leveraging the rancher monitoring chart and [kubevirt monitoring integration with the prometheus-operator](https://kubevirt.io/user-guide/operations/component_monitoring/#integrating-with-the-prometheus-operator)
- Add default Harvester monitoring dashboards, one for the dashboard page and the other for the VM instance details page.

**Testing Plan:**
1. harvester monitoring should be enabled by default when installed from the ISO
2. check if the monitoring pod are up and running
```
$kubectl get pod -n cattle-monitoring-system
NAME                                                     READY   STATUS    RESTARTS   AGE
prometheus-rancher-monitoring-prometheus-0               3/3     Running   1          3m47s
rancher-monitoring-grafana-7f54b7d8bc-8pj4c              3/3     Running   0          106s
rancher-monitoring-kube-state-metrics-744b9448f4-2tgwj   1/1     Running   0          4m4s
rancher-monitoring-operator-754bcd8cb4-w85jz             1/1     Running   0          4m4s
rancher-monitoring-prometheus-adapter-77568b975-k49n5    1/1     Running   0          4m4s
rancher-monitoring-prometheus-node-exporter-nsmzt        1/1     Running   0          4m4s
```
3. check if two built-in harvester monitoring dashboards are loaded via:
```
$kubectl get configmap -n cattle-dashboards | grep har
harvester-vm-dashboard                                 1      9m14s
harvester-vm-detail-dashboard                          1      9m14s
```
4. check prometheusrules and servicemonitors are exist
```
$ kubectl get servicemonitor -n cattle-monitoring-system
cattle-monitoring-system   prometheus-kubevirt-rules               28m
cattle-monitoring-system   rancher-monitoring-grafana              27m
cattle-monitoring-system   rancher-monitoring-kube-state-metrics   27m
cattle-monitoring-system   rancher-monitoring-node-exporter        27m
cattle-monitoring-system   rancher-monitoring-operator             27m
cattle-monitoring-system   rancher-monitoring-prometheus           27m

$kubectl get prometheusrules -n harvester-system
NAME                        AGE
prometheus-kubevirt-rules   29m
```
5. check graphs from the dashboard page and VM details page are showing correctly.
![image](https://user-images.githubusercontent.com/4569037/134501783-bd278285-1243-48ce-a642-5944ff9bd905.png)
